### PR TITLE
Promote image trigger annotation to GA and correct syntax

### PR DESCRIPTION
**Pull request by smarterclayton**
Use the correct JSONPath syntax (this may be something to backport to
3.6.1).

@bparees @mfojtik @soltysh

### CODE REVIEW

**Comment by bparees** [comment]
/hold
(we're post-devcut)

**Comment by smarterclayton** [comment]
Pointless to use hold like this.

On Sep 8, 2017, at 12:25 AM, Ben Parees <notifications@github.com> wrote:

/hold
(we're post-devcut)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/16226#issuecomment-327995517>,
or mute
the thread

**Comment by smarterclayton** [comment]
Oh, I didn't see you lgtm.  We need to talk about some migration implications first - removing label.

**Comment by bparees** [comment]
> Oh, I didn't see you lgtm.

yeah i was trying to get fancy.

> We need to talk about some migration implications first - removing label.

but it's alpha! :)

**Comment by mfojtik** [comment]
@smarterclayton what happen to existing resources with this annotation? will they stop working? or we don't care because it was alpha?

File: pkg/image/apis/image/v1/trigger/trigger.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package trigger
 // TriggerAnnotationKey is the annotation used on resources to signal they wish to have
 // container image references changed when an image stream tag is updated. Today, only
 // containers can be specified by fieldPath.
-const TriggerAnnotationKey = "image.alpha.openshift.io/triggers"
+const TriggerAnnotationKey = "image.openshift.io/triggers"
```
Comment (soltysh) [inline pkg/image/apis/image/v1/trigger/trigger.go]: I'll point out here, what @mfojtik pointed in the comments. Are you thinking of existing users? Do you plan some kind of migration/release notes about this change. Why not supporting both of the annotation for a release for a smooth migration and change only after?

File: pkg/image/trigger/annotations/annotations.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func findContainerBySelector(spec ometa.PodSpecReferenceMutator, init bool, sele
 		return spec.GetContainerByIndex(init, i)
 	}
 	// TODO: potentially make this more flexible, like whitespace
-	if name := strings.TrimSuffix(strings.TrimPrefix(selector, "?(@.name='"), "')"); name != selector {
+	if name := strings.TrimSuffix(strings.TrimPrefix(selector, "?(@.name==\""), "\")"); name != selector {
```
Comment (soltysh) [inline pkg/image/trigger/annotations/annotations.go]: I start to wonder what is worse, jsonpath or regexp, 🤔

**Comment by smarterclayton** [comment]
Jsonpath is worse because no one ever spec'd it.

On Sep 8, 2017, at 6:45 AM, Maciej Szulik <notifications@github.com> wrote:

*@soltysh* approved this pull request.

Overall lgtm, but I'd like a clear answer wrt to migration path.
------------------------------

In pkg/image/apis/image/v1/trigger/trigger.go
<https://github.com/openshift/origin/pull/16226#discussion_r137761582>:

> @@ -3,7 +3,7 @@ package trigger
 // TriggerAnnotationKey is the annotation used on resources to signal
they wish to have
 // container image references changed when an image stream tag is
updated. Today, only
 // containers can be specified by fieldPath.
-const TriggerAnnotationKey = "image.alpha.openshift.io/triggers"
+const TriggerAnnotationKey = "image.openshift.io/triggers"

I'll point out here, what @mfojtik <https://github.com/mfojtik> pointed in
the comments. Are you thinking of existing users? Do you plan some kind of
migration/release notes about this change. Why not supporting both of the
annotation for a release for a smooth migration and change only after?
------------------------------

In pkg/image/trigger/annotations/annotations.go
<https://github.com/openshift/origin/pull/16226#discussion_r137761914>:

> @@ -78,7 +78,7 @@ func findContainerBySelector(spec ometa.PodSpecReferenceMutator, init bool, sele
 		return spec.GetContainerByIndex(init, i)
 	}
 	// TODO: potentially make this more flexible, like whitespace
-	if name := strings.TrimSuffix(strings.TrimPrefix(selector,
"?(@.name='"), "')"); name != selector {
+	if name := strings.TrimSuffix(strings.TrimPrefix(selector,
"?(@.name==\""), "\")"); name != selector {

I start to wonder what is worse, jsonpath or regexp, 🤔

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/16226#pullrequestreview-61471626>,
or mute the thread

File: pkg/image/apis/image/v1/trigger/trigger.go (hunk 1 above)
Comment (smarterclayton) [reply pkg/image/apis/image/v1/trigger/trigger.go]: My current thinking is how #16225 does it

**Comment by smarterclayton** [comment]
After thinking about this more I don't think we should provide a migration.  Alpha is alpha, and this was only in 3.6.  The reason why serving certs deserve a migration of some form is that we are using them in deployed code and have been using them since 3.3.  This feature has a smaller window, and our general statement of alpha being unsupported and for preview stands.

**Comment by bparees** [comment]
I'm all for anything that sets a precedent like this :)

**Comment by soltysh** [comment]
> After thinking about this more I don't think we should provide a migration. Alpha is alpha, and this was only in 3.6. The reason why serving certs deserve a migration of some form is that we are using them in deployed code and have been using them since 3.3. This feature has a smaller window, and our general statement of alpha being unsupported and for preview stands.

How much more work is to provide the migration. I like precedents (sometimes), but at the same time I'd like to be nice (at least try) for our users and provide the with tooling for migration.

**Comment by smarterclayton** [comment]
Considering this landed in 3.6 (out for a few weeks) and that we are
considering backporting the fix, it's hard to say that any amount of work
is worth it.

If we're going to invest the effort, there are better places to do so.


On Sep 18, 2017, at 9:05 AM, Maciej Szulik <notifications@github.com> wrote:

After thinking about this more I don't think we should provide a migration.
Alpha is alpha, and this was only in 3.6. The reason why serving certs
deserve a migration of some form is that we are using them in deployed code
and have been using them since 3.3. This feature has a smaller window, and
our general statement of alpha being unsupported and for preview stands.

How much more work is to provide the migration. I like precedents
(sometimes), but at the same time I'd like to be nice (at least try) for
our users and provide the with tooling for migration.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/16226#issuecomment-330213363>,
or mute
the thread

**Comment by soltysh** [comment]
> Considering this landed in 3.6 (out for a few weeks) and that we are considering backporting the fix, it's hard to say that any amount of work is worth it. 

> If we're going to invest the effort, there are better places to do so.

OK, I think the last sentence convinced me more the the former 😉

**Pull request merged**